lms/djangoapps/course_api/blocks/forms.py

```diff
@@ -130,5 +99,72 @@ def clean(self):
         if not usage_key:
             return
 
-        cleaned_data['user'] = self.clean_requested_user(cleaned_data, usage_key.course_key)
+        cleaned_data['user'] = self._clean_requested_user(cleaned_data, usage_key.course_key)
         return cleaned_data
+
+    def _clean_requested_user(self, cleaned_data, course_key):
+        """
+        Validates and returns the requested_user, while checking permissions.
+        """
+        requesting_user = self.initial['requesting_user']
+        requested_username = cleaned_data.get('username', None)
+
+        if not requested_username:
+            return self._clean_no_user(requesting_user, cleaned_data, course_key)
+
+        elif requesting_user.username.lower() == requested_username.lower():
+            return self._clean_self_user(requesting_user, course_key)
+
+        else:
+            return self._clean_other_user(requesting_user, requested_username, course_key)
+
+    @staticmethod
+    def _clean_no_user(requesting_user, cleaned_data, course_key):
+        """
+        Verifies form for when no username is specified, including permissions.
+        """
+        # Verify that access to all blocks is requested
+        # (and not unintentionally requested).
+        if not cleaned_data.get('all_blocks', None):
+            raise ValidationError({'username': ['This field is required unless all_blocks is requested.']})
+
+        # Verify all blocks can be accessed for the course.
+        if not permissions.can_access_all_blocks(requesting_user, course_key):
+            raise PermissionDenied(
+                "'{requesting_username}' does not have permission to access all blocks in '{course_key}'."
+                .format(requesting_username=requesting_user.username, course_key=unicode(course_key))
+            )
+
+        # return None for user
+        return None
+
+    @staticmethod
+    def _clean_self_user(requesting_user, course_key):
```

Comment:
I found this method name a bit confusing because it has "self" in it and I think of that as being method on an instance, whereas this is static.  Perhaps `_clean_requesting_user ` would be clearer.